### PR TITLE
Swagger change for pre and post script

### PR DESCRIPTION
**Pull request by jemex**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by annatisch** [comment]
@jemex - this PR includes breaking changes yet does not alter the API version.
Were these changes actually made to the service without updating the API version? Or were these always errors in the spec? 

Also, please fix the semantic and model validation errors in CI:
```
 error : 
code: OBJECT_MISSING_REQUIRED_PROPERTY_DEFINITION
message: 'Missing required property definition: updateConfiguration'
path:
  - definitions
  - softareUpdateConfigurationRunTasks
jsonUrl: >-
  specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/softwareUpdateConfigurationRun.json


 error : 
code: OBJECT_MISSING_REQUIRED_PROPERTY_DEFINITION
message: 'Missing required property definition: scheduleInfo'
path:
  - definitions
  - softareUpdateConfigurationRunTasks
jsonUrl: >-
  specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/softwareUpdateConfigurationRun.json
```

**Comment by jemex** [comment]
@annatisch, this PR doesn't have breaking changes. we only added new properties to the existing model.

**Comment by jemex** [comment]
@annatisch - Thank you so much for the feedback. I have updated the "required" field.  the checks takes longer to validate them, what do I need to do?

**Comment by jemex** [comment]
@annatisch- can you merge this PR please if you it looks ok to you.

**Comment by annatisch** [comment]
@jemex - Can you please fix the model validation errors? I will review once CI is clean:
```
 error : 
operationId: SoftwareUpdateConfigurations_Create
source: response
responseCode: '200'
errorDetails:
  code: OBJECT_ADDITIONAL_PROPERTIES
  params:
    - - locations
  message: 'Additional properties not allowed: locations'
  jsonUrl: >-
    specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/examples/softwareUpdateConfiguration/createSoftwareUpdateConfiguration.json

 error : 
operationId: SoftwareUpdateConfigurations_Create
source: response
responseCode: '201'
errorDetails:
  code: OBJECT_ADDITIONAL_PROPERTIES
  params:
    - - locations
  message: 'Additional properties not allowed: locations'
  jsonUrl: >-
    specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/examples/softwareUpdateConfiguration/createSoftwareUpdateConfiguration.json

 error : 
operationId: SoftwareUpdateConfigurations_GetByName
source: response
responseCode: '200'
errorDetails:
  code: OBJECT_ADDITIONAL_PROPERTIES
  params:
    - - locations
  message: 'Additional properties not allowed: locations'
  jsonUrl: >-
    specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/examples/softwareUpdateConfiguration/getSoftwareUpdateConfigurationByName.json
```

**Review by annatisch — CHANGES_REQUESTED** [review]
Looks pretty good - just a couple of minor things to improve the generated code :)

File: specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/softwareUpdateConfiguration.json [hunk 1]
```diff
@@ -781,7 +816,7 @@
             "description": "Subscription or Resource Group ARM Id."
           }
         },
-        "location": {
+        "locations": {
```
[the remark this answers is not in the document]
Comment (jemex) [reply specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/softwareUpdateConfiguration.json]: I added that property last week and I am missed 's'

File: specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/softwareUpdateConfiguration.json [hunk 2]
```diff
@@ -348,15 +348,50 @@
...
+          "description": "Tasks information for the Software update configuration.",
+          "$ref": "#/definitions/tasksProperties"
         }
       },
       "required": [
         "updateConfiguration",
         "scheduleInfo"
       ]
     },
+    "tasksProperties": {
```
[the remark this answers is not in the document]
Comment (jemex) [reply specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/softwareUpdateConfiguration.json]: renamed to softwareUpdateConfigurationTasks

File: specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/softwareUpdateConfigurationRun.json [hunk 3]
```diff
@@ -260,8 +260,44 @@
...
+    "softareUpdateConfigurationRunTasks": {
+      "description": "Software update configuration run tasks model.",
+      "type": "object",
+      "properties": {
+        "preTask": {
+          "description": "Pre task object.",
+          "$ref": "#/definitions/softareUpdateConfigurationRunTaskProperties"
+        },
+        "postTask": {
+          "description": "Post task object.",
```
[the remark this answers is not in the document]
Comment (annatisch) [reply specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/softwareUpdateConfigurationRun.json]: @jemex - you missed this one :)

**Comment by jemex** [comment]
@annatisch - I have resolved the feedback. Can you tell me if I have missed anything? If not and if it  looks good, can you merge , please?

**Comment by annatisch** [comment]
Thanks @jemex 
You missed one:
https://github.com/Azure/azure-rest-api-specs/pull/4298/files#diff-78182b2e6ebf616ace2b0ed13f9e4c96R281

**Comment by annatisch** [comment]
@AutorestCI regenerate azure-sdk-for-python

**Review by annatisch — APPROVED** [review]

**Pull request merged**